### PR TITLE
Flag `I` as a conflicting variable name

### LINKED ISSUE #1998: Bug when compiling with C when using variable `I`
### Describe the bug
When attempting to compile Python scripts using Pyccel with C, the process fails during the shared library generation stage when using the variable `I`. The error appears to be related to the handling of global variables in the generated C code. Could it be related to the imaginary unit `I`?

### To Reproduce

1. Clone git repo, setup virtual environment, and install Pyccel

```bash
git clone --recurse-submodules git@github.com:pyccel/pyccel.git
cd pyccel

python3 -m venv env_pyccel
source env_pyccel/bin/activate
pip install .
```
2. Create two simple Python files (`test_I.py` and `test_J.py`) that define variables `J` and `I`, respectively.

```
echo "J=1" > test_J.py
echo "I=1" > test_I.py
```

3. Compile test_I.py and test_J.py with fortran and C.

```
pyccel test_J.py --language=fortran
pyccel test_J.py --language=c
pyccel test_I.py --language=fortran
pyccel test_I.py --language=c
```

The three first tests work fine for me, but I get an error on the fourth.

### Error details
The error occurs specifically when compiling `test_I.py` with C.

Error messages:

```bash
> pyccel test_J.py --language=fortran
> pyccel test_J.py --language=c
> pyccel test_I.py --language=fortran
> pyccel test_I.py --language=c

ERROR at shared library generation stage
Traceback (most recent call last):
  File "/home/maxlinadmin/git_repos/pyccel/env_pyccel/bin/pyccel", line 8, in <module>
    sys.exit(pyccel())
             ^^^^^^^^
  File "/home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/pyccel/commands/console.py", line 266, in pyccel
    execute_pyccel(filename,
  File "/home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/pyccel/codegen/pipeline.py", line 408, in execute_pyccel
    generated_filepath, shared_lib_timers = create_shared_library(codegen,
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/pyccel/codegen/python_wrapper.py", line 220, in create_shared_library
    wrapper_compiler.compile_module(wrapper_compile_obj,
  File "/home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/pyccel/codegen/compiling/compilers.py", line 358, in compile_module
    self.run_command(cmd, verbose)
  File "/home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/pyccel/codegen/compiling/compilers.py", line 503, in run_command
    raise RuntimeError(err_msg)
RuntimeError: Failed to build module
In file included from /home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/numpy/_core/include/numpy/npy_common.h:380,
                 from /home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarraytypes.h:4,
                 from /home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:12,
                 from /home/maxlinadmin/git_repos/pyccel/env_pyccel/lib/python3.12/site-packages/numpy/_core/include/numpy/arrayobject.h:5,
                 from /home/maxlinadmin/git_repos/pyccel/__pyccel__/test_I_wrapper.h:5,
                 from /home/maxlinadmin/git_repos/pyccel/__pyccel__/test_I_wrapper.c:4:
/home/maxlinadmin/git_repos/pyccel/__pyccel__/test_I.h:8:16: error: expected identifier or ‘(’ before ‘__extension__’
    8 | extern int64_t I;
      |                ^
/home/maxlinadmin/git_repos/pyccel/__pyccel__/test_I_wrapper.c: In function ‘PyInit_test_I’:
/home/maxlinadmin/git_repos/pyccel/__pyccel__/test_I_wrapper.c:56:34: error: lvalue required as unary ‘&’ operand
   56 |     Dummy_0000 = Int64_to_PyLong(&I);
      |                                  ^

```

### Expected behavior
The Python scripts `test_I.py` and `test_J.py` should be successfully compiled using Pyccel both with Fortran and C.

### Language

Error occurs only in C.

### Additional context

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add `I` to the list of names of variables which are not valid in C. Fixes #1998

### CODE REVIEW

**Review by yguclu — APPROVED** [review]

File: pyccel/naming/languagenameclashchecker.py [hunk 1]
```diff
@@ -19,6 +19,11 @@ class LanguageNameClashChecker(metaclass = Singleton):
     """
     keywords = None
 
+    def __init__(self): #pylint: disable=useless-parent-delegation
+        # This __init__ function is required so the Singleton can detect a signature
+        super().__init__()
+
+
     def _get_collisionless_name(self, name, symbols):
```
Comment (yguclu) [inline pyccel/naming/languagenameclashchecker.py:27]: Too many blank lines to separate methods of a class. One is enough 😉

Comment (EmilyBourne) [reply pyccel/naming/languagenameclashchecker.py]: Fixed [e0f8120](https://github.com/pyccel/pyccel/pull/2003/commits/e0f8120fd44b8d808bc17a1af4c6fdc920cf9a87)

**Review by yguclu — APPROVED** [review]

**Pull request merged**